cli/test/e2e/package_repository_test.go

```diff
@@ -181,6 +186,46 @@ func TestPackageRepository(t *testing.T) {
 		require.Exactly(t, expectedOutputRows, output.Tables[0].Rows)
 	})
 
+	logger.Section("create a repository with secretRef", func() {
+		_, _ = kappCtrl.RunWithOpts([]string{"package", "repository", "add", "-r", pkgrWithSecretName, "--url", pkgrURL, "--secret-ref", pkgrSecretRef}, RunOpts{
+			AllowError: true})
+
+		pkgrYaml := kubectl.Run([]string{"get", kind, pkgrWithSecretName, "-ojson"})
+		pkgr := &kcpkg.PackageRepository{}
+		err := json.Unmarshal([]byte(pkgrYaml), pkgr)
+		require.NoError(t, err)
+		require.Equal(t, pkgrSecretRef, pkgr.Spec.Fetch.ImgpkgBundle.SecretRef.Name)
+
+		kappCtrl.Run([]string{"package", "repository", "delete", "-r", pkgrWithSecretName})
+	})
+
+	logger.Section("updating a repository's secret with no change in url", func() {
+		_, err := kappCtrl.RunWithOpts([]string{"package", "repository", "add", "-r", pkgrWithSecretName, "--url", pkgrURL}, RunOpts{
```

Comment:
I do not think we need to create another pkgr, we can just use the previous pkgr, and not delete it in the previous section